ledger/metrics/src/main/scala/com/daml/metrics/HttpJsonApiMetrics.scala

```diff
@@ -25,43 +30,78 @@ class HttpJsonApiMetrics(val prefix: MetricName, override val registry: MetricRe
   // Meters how long a find by contract id database operation takes
-  val dbFindByContractId: Timer = timer(prefix :+ "db_find_by_contract_id_timing")
+  val dbFindByContractId: Timer =
+    DropWizardMetricsFactory.timer(prefix :+ "db_find_by_contract_id_timing")
   // Meters how long processing of the command submission request takes on the ledger
   val commandSubmissionLedgerTimer: Timer =
-    timer(prefix :+ "command_submission_ledger_timing")
+    DropWizardMetricsFactory.timer(prefix :+ "command_submission_ledger_timing")
   // Meters http requests throughput
-  val httpRequestThroughput: Meter = meter(prefix :+ "http_request_throughput")
+  val httpRequestThroughput: Meter =
+    DropWizardMetricsFactory.meter(prefix :+ "http_request_throughput")
   // Meters how many websocket connections are currently active
-  val websocketRequestCounter: Counter = counter(prefix :+ "websocket_request_count")
+  val websocketRequestCounter: Counter =
+    DropWizardMetricsFactory.counter(prefix :+ "websocket_request_count")
   // Meters command submissions throughput
   val commandSubmissionThroughput: Meter =
-    meter(prefix :+ "command_submission_throughput")
+    DropWizardMetricsFactory.meter(prefix :+ "command_submission_throughput")
   // Meters package uploads throughput
-  val uploadPackagesThroughput: Meter = meter(prefix :+ "upload_packages_throughput")
+  val uploadPackagesThroughput: Meter =
+    DropWizardMetricsFactory.meter(prefix :+ "upload_packages_throughput")
   // Meters party allocation throughput
-  val allocatePartyThroughput: Meter = meter(prefix :+ "allocation_party_throughput")
+  val allocatePartyThroughput: Meter =
+    DropWizardMetricsFactory.meter(prefix :+ "allocation_party_throughput")
+
+  object OpenTelemetryMetricsFactory extends OpenTelemetryFactory {
+    override val otelMeter: OtelMeter = HttpJsonApiMetrics.this.otelMeter
+  }
+
+  // golden signals
+  val httpRequestsTotal: Counter = OpenTelemetryMetricsFactory.counter(prefix :+ "requests_total")
+  val httpErrorsTotal: Counter = OpenTelemetryMetricsFactory.counter(prefix :+ "errors_total")
+  val httpLatency: Timer = OpenTelemetryMetricsFactory.timer(prefix :+ "requests_duration_seconds")
+  val httpRequestsBytesTotal: Counter =
+    OpenTelemetryMetricsFactory.counter(prefix :+ "requests_bytes_total")
+  val httpResponsesBytesTotal: Counter =
+    OpenTelemetryMetricsFactory.counter(prefix :+ "responses_bytes_total")
+
+  val websocketReceivedTotal: Counter =
+    OpenTelemetryMetricsFactory.counter(prefix :+ "websocket_messages_received_total")
+  val websocketReceivedBytesTotal: Counter =
+    OpenTelemetryMetricsFactory.counter(prefix :+ "websocket_messages_received_bytes_total")
+  val websocketSentTotal: Counter =
+    OpenTelemetryMetricsFactory.counter(prefix :+ "websocket_messages_sent_total")
+  val websocketSentBytesTotal: Counter =
+    OpenTelemetryMetricsFactory.counter(prefix :+ "websocket_messages_sent_bytes_total")
```

Comment:
This is not what I am seeing.
This is a small part of the data available on the Prometheus end point. There are no extra `total` or `second` added, only the ones which are part of the metric name.
```
daml_http_json_api_requests_duration_seconds_bucket{daml_version="0.0.0",host="localhost",http_verb="GET",path="/v1/user",le="500.0",} 6.0 1667495595209
daml_http_json_api_requests_duration_seconds_bucket{daml_version="0.0.0",host="localhost",http_verb="GET",path="/v1/user",le="750.0",} 6.0 1667495595209
daml_http_json_api_requests_duration_seconds_bucket{daml_version="0.0.0",host="localhost",http_verb="GET",path="/v1/user",le="1000.0",} 6.0 1667495595209
daml_http_json_api_requests_duration_seconds_bucket{daml_version="0.0.0",host="localhost",http_verb="GET",path="/v1/user",le="2500.0",} 6.0 1667495595209
daml_http_json_api_requests_duration_seconds_bucket{daml_version="0.0.0",host="localhost",http_verb="GET",path="/v1/user",le="5000.0",} 6.0 1667495595209
daml_http_json_api_requests_duration_seconds_bucket{daml_version="0.0.0",host="localhost",http_verb="GET",path="/v1/user",le="7500.0",} 6.0 1667495595209
daml_http_json_api_requests_duration_seconds_bucket{daml_version="0.0.0",host="localhost",http_verb="GET",path="/v1/user",le="10000.0",} 6.0 1667495595209
daml_http_json_api_requests_duration_seconds_bucket{daml_version="0.0.0",host="localhost",http_verb="GET",path="/v1/user",le="+Inf",} 6.0 1667495595209
# HELP daml_http_json_api_websocket_messages_received_bytes_total 
# TYPE daml_http_json_api_websocket_messages_received_bytes_total gauge
daml_http_json_api_websocket_messages_received_bytes_total{daml_version="0.0.0",host="localhost",path="/v1/stream/query",} 240.0 1667495595209
# HELP daml_http_json_api_errors_total 
# TYPE daml_http_json_api_errors_total gauge
daml_http_json_api_errors_total{daml_version="0.0.0",host="localhost",http_status="404",http_verb="POST",path="/v1/create",} 8384.0 1667495595209
daml_http_json_api_errors_total{daml_version="0.0.0",host="localhost",http_status="404",http_verb="POST",path="/v1/exercise",} 8369.0 1667495595209
daml_http_json_api_errors_total{daml_version="0.0.0",host="localhost",http_status="400",http_verb="POST",path="/v1/exercise",} 8369.0 1667495595209
# HELP daml_http_json_api_websocket_messages_received_total 
# TYPE daml_http_json_api_websocket_messages_received_total gauge
daml_http_json_api_websocket_messages_received_total{daml_version="0.0.0",host="localhost",path="/v1/stream/query",} 6.0 1667495595209
# HELP daml_http_json_api_requests_bytes_total 
# TYPE daml_http_json_api_requests_bytes_total gauge
daml_http_json_api_requests_bytes_total{daml_version="0.0.0",host="localhost",http_verb="POST",path="/v1/packages",} 315684.0 1667495595209
daml_http_json_api_requests_bytes_total{daml_version="0.0.0",host="localhost",http_verb="POST",path="/v1/create",} 1465787.0 1667495595209
daml_http_json_api_requests_bytes_total{daml_version="0.0.0",host="localhost",http_verb="POST",path="/v1/exercise",} 2942540.0 1667495595209
daml_http_json_api_requests_bytes_total{daml_version="0.0.0",host="localhost",http_verb="GET",path="/v1/stream/query",} 0.0 1667495595209
daml_http_json_api_requests_bytes_total{daml_version="0.0.0",host="localhost",http_verb="POST",path="/v1/user/create",} 258.0 1667495595209

Reply:
@nicu-da Can you possibly point us to the documentation that describes the behaviour you are expecting (i.e., addition of `total` and `seconds` suffixes), please? I'm wondering if we are configuring the exporter correctly.

Reply:
Did we try without the `_total` suffix? This naming is prometheus specific, we should not add it in our code.
Looking at the prometheus client code, they actually remove the suffix if it's set, as it's set at the final export step.

Reply:
Currently, it does not get automagically added.
Here the metrics data after I removed the `_total` and `_seconds` suffixes:
```
daml_http_json_api_requests_duration_bucket{daml_version="0.0.0",host="localhost",http_verb="GET",path="/v1/user",le="250.0",} 1.0 1667554914673
daml_http_json_api_requests_duration_bucket{daml_version="0.0.0",host="localhost",http_verb="GET",path="/v1/user",le="500.0",} 1.0 1667554914673
daml_http_json_api_requests_duration_bucket{daml_version="0.0.0",host="localhost",http_verb="GET",path="/v1/user",le="750.0",} 1.0 1667554914673
daml_http_json_api_requests_duration_bucket{daml_version="0.0.0",host="localhost",http_verb="GET",path="/v1/user",le="1000.0",} 1.0 1667554914673
daml_http_json_api_requests_duration_bucket{daml_version="0.0.0",host="localhost",http_verb="GET",path="/v1/user",le="2500.0",} 1.0 1667554914673
daml_http_json_api_requests_duration_bucket{daml_version="0.0.0",host="localhost",http_verb="GET",path="/v1/user",le="5000.0",} 1.0 1667554914673
daml_http_json_api_requests_duration_bucket{daml_version="0.0.0",host="localhost",http_verb="GET",path="/v1/user",le="7500.0",} 1.0 1667554914673
daml_http_json_api_requests_duration_bucket{daml_version="0.0.0",host="localhost",http_verb="GET",path="/v1/user",le="10000.0",} 1.0 1667554914673
daml_http_json_api_requests_duration_bucket{daml_version="0.0.0",host="localhost",http_verb="GET",path="/v1/user",le="+Inf",} 1.0 1667554914673
# HELP daml_http_json_api_requests 
# TYPE daml_http_json_api_requests gauge
daml_http_json_api_requests{daml_version="0.0.0",host="localhost",http_status="404",http_verb="GET",path="/v1/user",} 1.0 1667554914673
daml_http_json_api_requests{daml_version="0.0.0",host="localhost",http_status="404",http_verb="POST",path="/v1/create",} 84.0 1667554914673
daml_http_json_api_requests{daml_version="0.0.0",host="localhost",http_verb="GET",path="/v1/stream/query",} 1.0 1667554914673
daml_http_json_api_requests{daml_version="0.0.0",host="localhost",http_status="404",http_verb="POST",path="/v1/exercise",} 162.0 1667554914673
daml_http_json_api_requests{daml_version="0.0.0",host="localhost",http_status="200",http_verb="GET",path="/readyz",} 1.0 1667554914673
# HELP daml_http_json_api_errors 
# TYPE daml_http_json_api_errors gauge
daml_http_json_api_errors{daml_version="0.0.0",host="localhost",http_status="404",http_verb="GET",path="/v1/user",} 1.0 1667554914673
daml_http_json_api_errors{daml_version="0.0.0",host="localhost",http_status="404",http_verb="POST",path="/v1/create",} 84.0 1667554914673
daml_http_json_api_errors{daml_version="0.0.0",host="localhost",http_verb="GET",path="/v1/stream/query",} 1.0 1667554914673
daml_http_json_api_errors{daml_version="0.0.0",host="localhost",http_status="404",http_verb="POST",path="/v1/exercise",} 162.0 1667554914673
# HELP daml_http_json_api_requests_bytes 
# TYPE daml_http_json_api_requests_bytes gauge
daml_http_json_api_requests_bytes{daml_version="0.0.0",host="localhost",http_verb="POST",path="/v1/create",} 8610.0 1667554914673
daml_http_json_api_requests_bytes{daml_version="0.0.0",host="localhost",http_verb="POST",path="/v1/exercise",} 22330.0 1667554914673
daml_http_json_api_requests_bytes{daml_version="0.0.0",host="localhost",http_verb="GET",path="/v1/stream/query",} 0.0 1667554914673
```

Reply:
@lucb-da The problem is this: 
```# TYPE daml_http_json_api_errors gauge``` it should not be a gauge

Reply:
@nicu-da 
* ok, so it works for `_total` if get it as a counter to Prometheus. But there is a big discrepancy between the Prometheus naming convention and the  types in `MetricHande`:
  * `MetricHandle` `Meter` => Prometheus counter
  * `MetricHandle` `Counter` => Prometheus up/down gauge
  * `MetricHandle` `Gauge` => Prometheus 'set value' gauge
  We likely need a big warning in `MetricHandle`, otherwise others will make the same mistake.
* Is there a trick to automatically get the`_seconds` for the latency ?
  The definition uses the `Timer` type from `MetricHande`
  It appears like this in the Prometheus endpoint
```
# HELP daml_http_json_api_requests_duration_seconds 
# TYPE daml_http_json_api_requests_duration_seconds histogram
daml_http_json_api_requests_duration_seconds_count{daml_version="0.0.0",host="localhost",http_verb="POST",path="/v1/create",} 66.0 1667832008030
daml_http_json_api_requests_duration_seconds_sum{daml_version="0.0.0",host="localhost",http_verb="POST",path="/v1/create",} 4721.0 1667832008030
daml_http_json_api_requests_duration_seconds_bucket{daml_version="0.0.0",host="localhost",http_verb="POST",path="/v1/create",le="5.0",} 0.0 1667832008030
daml_http_json_api_requests_duration_seconds_bucket{daml_version="0.0.0",host="localhost",http_verb="POST",path="/v1/create",le="10.0",} 0.0 1667832008030
daml_http_json_api_requests_duration_seconds_bucket{daml_version="0.0.0",host="localhost",http_verb="POST",path="/v1/create",le="25.0",} 0.0 1667832008030
```
* And in general, I am not sure about removing the suffixes. The Prometheus exporter does automatically add them, but other exporters might not. Having them all the time is useful, and the Prometheus exporter seems to manage if they are already there.

Reply:
Oh, really good catch. It makes total sense though as a counter as defined by prometheus should be monotonically increasing and in our case the `Counter` maps to the definition of a gauge. 
We can keep it as is for now, but we should probably clarify our API.